client/gutenberg/extensions/publicize/form-unwrapped.jsx

```diff
@@ -44,14 +50,19 @@ class PublicizeFormUnwrapped extends Component {
 		return ! formEnabled;
 	}
 
+	getShareMessage() {
+		const { shareMessage, defaultShareMessage } = this.props;
+		return this.state.hasEditedShareMessage ? shareMessage : defaultShareMessage;
```

Comment:
I don't think you actually need component state (`hasEditedShareMessage`) to determine what to display here. The information whether the message has been manually edited should be fully contained in `shareMessage` and `defaultShareMessage`. You can probably just

```jsx
return shareMessage !== defaultShareMessage ? shareMessage : '';
```

and drop `onMessageChange` altogether (just reverting to using `messageChange` for the `textarea`'s `onChange` prop.

Reply:
^ This should also fix the [bug](https://github.com/Automattic/wp-calypso/pull/28695#pullrequestreview-176525372) I found

Reply:
I pushed a change in logic that fixes this bug in 4c78a14465f8319bcff0cd21eec908e7bfa06c4f. Your suggestion would work nicely if we decided to use the post title as the placeholder. I'd like to keep a more descriptive placeholder, so i had to shake up the logic a bit!

Reply:
Yeah, I think it's not so much about the placeholder content rather than about different behavior upon first load -- it's true that with my suggestion, we'd be displaying the placeholder on first load (rather than the default message).